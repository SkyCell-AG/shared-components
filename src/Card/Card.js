import React from 'react'
import {
    Card as MaterialCard,
    CardContent,
    Typography,
    Icon,
} from '@mui/material'
import PropTypes from 'prop-types'
import clsx from 'clsx'

import {
    PENDING,
} from 'utils/requestStatuses'
import Loading from 'Loading'

import useStyles from './Card.style'

const propTypes = {
    children: PropTypes.node.isRequired,
    title: PropTypes.string,
    icon: PropTypes.string,
    className: PropTypes.string,
    contentClass: PropTypes.string,
    titleClass: PropTypes.string,
    fullHeight: PropTypes.bool,
    status: PropTypes.string,
}

const defaultProps = {
    className: '',
    contentClass: '',
    titleClass: '',
    title: '',
    icon: '',
    fullHeight: false,
    status: '',
}

const Card = (props) => {
    const classes = useStyles()
    const {
        children,
        title,
        icon,
        className,
        contentClass,
        titleClass,
        fullHeight,
        status,
    } = props

    return (
        <MaterialCard
            className={clsx(classes.card, className, {
                [classes.fullHeight]: fullHeight,
            })}
        >
            <CardContent className={contentClass}>
                {title
                    && (
                        <Typography className={clsx(classes.title, titleClass)}>
                            {title}
                            {icon
                                && (
                                    <Icon className={classes.icon}>{icon}</Icon>
                                )}
                            {status === PENDING
                                && (
                                    <div className={classes.loading}>
                                        <Loading
                                            size={15}
                                            color="inherit"
                                        />
                                    </div>
                                )}
                        </Typography>
                    )}
                {children}
            </CardContent>
        </MaterialCard>
    )
}

Card.propTypes = propTypes
Card.defaultProps = defaultProps

export default Card
