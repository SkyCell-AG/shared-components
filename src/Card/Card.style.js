import {
    makeStyles,
} from '@mui/styles'
import {
    useTheme,
} from '@mui/material/styles'

const useStyles = makeStyles(() => {
    const theme = useTheme()

    return {
        card: {
            background: theme.palette.common.white,
            boxShadow: `0px 1px 3px ${theme.palette.secondary.main}`,
            borderRadius: '4px',
            height: 'auto',
            overflow: 'visible !important',
            position: 'relative',
        },
        contentClass: {
            padding: 10,
        },
        fullHeight: {
            height: '100%',
        },
        title: {
            fontWeight: '400',
            fontSize: '20px !important',
            letterSpacing: '0.26px',
            color: theme.palette.secondary.dark,
            display: 'flex',
        },
        icon: {
            padding: '5px 0 0 5px',
            width: '30px',
            height: '26px',
            position: 'relative',
            top: '4px',
        },
        loading: {
            position: 'relative',
            marginLeft: 6,
            width: 15,
        },
    }
})

export default useStyles
