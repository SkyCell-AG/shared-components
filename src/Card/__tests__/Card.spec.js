import React from 'react'
import {
    shallow,
} from 'enzyme'

import Card from '../Card'

it('Card snapshot', () => {
    const wrapper = shallow(
        <Card>
            TEST
        </Card>,
    )

    expect(wrapper).toMatchSnapshot()
})
