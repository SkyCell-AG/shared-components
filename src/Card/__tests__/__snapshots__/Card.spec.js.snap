// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot 1`] = `
<ForwardRef(Card)
  className="makeStyles-card-1"
>
  <ForwardRef(CardContent)
    className="makeStyles-contentClass-3"
  >
    TEST
  </ForwardRef(CardContent)>
</ForwardRef(Card)>
`;
